vulcanai2/tests/models/test_dnn.py

```diff
@@ -37,13 +51,38 @@ def test_init(self, dnn_noclass):
         assert not hasattr(dnn_noclass, 'metrics')
 
     def test_function_multi_input(self, multi_input_dnn):
-        """Test methods/functions wrt multi_input_cnn"""
+        """Test functions wrt multi_input_dnn"""
         assert isinstance(multi_input_dnn.input_networks, nn.ModuleDict)
         assert len(list(multi_input_dnn.input_networks)) == 2
         assert multi_input_dnn._merge_input_network_outputs([
                                 torch.ones([10, 1, 28]),
                                 torch.ones([10, 1, 28, 28])
                ]).shape == (10, 812)
+    
+    def test_forward(self, dnn_class):
+        """Test Forward of DenseNet"""
+        out = dnn_class(torch.ones([10, *dnn_class.in_dim]))
+        assert out.shape == (10, 3)
+    
+    def test_forward_multi_input(self, multi_input_dnn):
+        """Test Forward of Multi Input ConvNet"""
+        master_device_setter(multi_input_dnn, 'cuda:0')
+        input_tensor = [torch.ones([10, 1, 28]),
+                        torch.ones([10, 1, 28, 28])]
+        out = multi_input_dnn(input_tensor)
+        assert out.shape == (10, 50)
+
+    # TODO: Failing! Fix the test
+    def test_forward_multi_input_dnn_add_input_network(self, 
+                                multi_input_dnn_add_input_network):
+        """Test Forward of Multi Input DenseNet where input_networks
+        added via add_input_network"""
+        out = multi_input_dnn_add_input_network([
+                            torch.ones([10, 1, 28]),
+                            torch.ones([10, 1, 28, 28]),
+                            torch.ones([10, 200])
+                        ])
```

Comment:
@rfratila and this one fails. I think we need to replace the first layer of the basenetworks's`.network` with the updated `self._get_in_dim()` after adding the `input_network` via `add_input_network`

Reply:
I believe this would solve the issue we had with the cnn as well earlier

Reply:
For now @priyatharsan turn `add_input_network` into a class private `_add_input_network` function to avoid adding a network after it's been created until we come up with a better solution. For now it's simpler if an input_network is specified at construction and not afterworlds. We won't run into these issues

Reply:
do not replace any layer.